### DISCUSSION TITLE
How to include blog front matter images in rss feeds?

### DISCUSSION THREAD

**Discussion by r3dcrosse** [facebook/docusaurus#8321] [Q&A] [answered]
We've gotten great use out of Docusaurus, thank you for making such an amazing and highly customizable project. 

We have a particular use case with rss feeds Docusaurus produces and I've been wondering, what is the best way to get an image from front matter included in rss feeds?

For example, with a front matter like:
```
---
title: Blog post title
description: A description here
authors: anauthor
image: /img/blog_headers/CarBlogCoverMini.jpg
---
```
I'd like to get an rss.xml like:
```
<item>
  <title>
    ...
  </title>
  <description>
  ...
  </description>
  <content:encoded>
  ...
  </content:encoded>
  <enclosure url="https://myblogsite/img/blog_headers/CarBlogCoverMini.jpg" length="0" type="image/jpg"/>
</item>
```
and feed.json like:
```
{
            ...
            "title": "Blog post title",
            "summary": "A description here.",
            "image": "https://myblogsite/img/blog_headers/CarBlogCoverMini.jpg",
}
```

Would this be possible by building a separate plugin to handle generating the rss/json feeds?

For now, I've hacked together a rickety solution by modifying the blog plugin to create the feed item with the image. My hack worked wonderfully when using webpack4, but after upgrading to webpack5, required me to stop using relative paths in my front matter, and use the static assets image path instead.

**Comment by slorber** [comment, marked as answer]
Hey

In my case the blog image social card displays properly on Feedly, despite the lack of this json/rss attribute.

<img width="952" alt="CleanShot 2022-11-17 at 11 50 49@2x" src="https://user-images.githubusercontent.com/749374/202427238-2b586356-681d-4d24-8146-e770c14878ed.png">

What's your RSS reader? Can't it read the `og:image` tag of the linked page?

--- 

That's probably worth opening a feature request, it should be quite easy to fix and our feed npm lib supports adding such extra attribute.

This is pseudo-code and I didn't run it but could be as simple as:

<img width="588" alt="CleanShot 2022-11-17 at 11 54 07@2x" src="https://user-images.githubusercontent.com/749374/202427875-74f87ddb-9b2f-4e73-b5c5-6f4bf913cdfa.png">


We could also read `blogPost.metadata.frontMatter.image` but I'm not sure it will work with relative image paths that are processed/hashed by Webpack. The final image URL is currently inaccessible on the nodejs side afaik.

**Reply by r3dcrosse** [reply]
Oh this is a brilliant idea! I would have never thought to make use of cheerio to get the image URL from the `og:image` meta tag.

This even works for my use case of using a relative image path in the blog post front matter, since the meta tag has the processed/hashed image final URL.

I would love to make a pull request that adds this! Should I still open a feature request for it?

**Reply by slorber** [reply]
That's a tiny improvement/fix so a direct PR is perfectly fine for that 👍  thanks

**Reply by lkzwc** [reply]
![image](https://user-images.githubusercontent.com/84896877/203207681-f6dc7f8c-f9ce-4baa-9422-61cd5712fb6c.png)

official has provide this template ?

**Reply by slorber** [reply]
> official has provide this template ?

I don't understand sorry 😅

**Comment by Fashander** [comment]
Hi there, 

Great work on Docusaurus!

We've been using it for our documentation and blog,  and its been great so far. 

Recently, we've been having the same issue as the one in this discussion and I wonder if the issue with the front matter image for RSS feeds has been resolved. 

Our situation is similar to the one described above and we've not been able to get blogpost images to display in our external marketing sites.

**Reply by slorber** [reply]
Docusaurus v2.3 introduced a `createFeedItem` config hook to let you override the default Docusaurus behavior: https://docusaurus.io/blog/releases/2.3#blog-createfeeditems

This gives you the opportunity to change the list of feed items, adjust the feed items themselves if needed, or totally take-over the creation of a feed item.

The item you choose to return must be compliant with our TS types, or what this library supports: https://github.com/jpmonette/feed
(unfortunately it's not super well documented so you'll have to look at the code or types for advanced cases like enclosures etc)

**Reply by Fashander** [reply]
Thanks @slorber . Your suggestions worked perfectly!
